do-not-merge: enable DRPC tests

cthumuluru-crdb opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153670)
<!-- Reviewable:end -->

cthumuluru-crdb committed:
server: force gRPC use to join the cluster

DRPC support for init service is not available yet. Force gRPC use to join the
cluster as a workaround.

cthumuluru-crdb committed:
rpc: enable DRPC by flipping the cluster setting
